storage/Utils/CommentedConfigFile.cc

```diff
@@ -0,0 +1,366 @@
+        //   first content line, the empty lines belong to the header comment.
+        //
+        // - If there were empty lines and then some more comment lines before
+        //   the first content line, the comments after the last empty line no
+        //   longer belong to the header comment, but to the first content
+        //   entry. So let's go back to that last empty line.
+    }
+
+    return header_end;
+}
+
+
+int CommentedConfigFile::find_footer_comment_start( const string_vec & lines,
+                                                    int from )
+{
+    int footer_start = -1;
+
+    for ( int i = lines.size()-1; i >= from; --i )
+    {
+        const string & line = lines[i];
+
+        if ( is_empty_line( line ) || is_comment_line( line ) )
+            footer_start = i;
+        else // found the last content line
+            break;
+    }
+
+    return footer_start;
+}
+
+
+string_vec CommentedConfigFile::format_lines()
+{
+    string_vec lines = header_comments;
+
+    for ( size_t i=0; i < entries.size(); ++i )
+    {
+        Entry * entry = entries[i];
+
+        for ( size_t j=0; j < entry->get_comment_before().size(); ++j )
+            lines.push_back( entry->get_comment_before()[j] );
+
+        string line = entry->format();
+
+        if ( ! entry->get_line_comment().empty() )
+            line += " " + entry->get_line_comment();
+
+        lines.push_back( line );
+    }
+
+    for ( size_t i=0; i < footer_comments.size(); ++i )
+        lines.push_back( footer_comments[i] );
+
+    return lines;
+}
+
+
+void CommentedConfigFile::clear_entries()
+{
+    for ( size_t i=0; i < entries.size(); ++i )
+	delete entries[i];
+
+    entries.clear();
+}
+
+
+void CommentedConfigFile::clear_all()
+{
+    clear_entries();
+    header_comments.clear();
+    footer_comments.clear();
+}
+
+
+bool CommentedConfigFile::is_comment_line( const string & line )
+{
+    size_t pos = line.find_first_not_of( WHITESPACE );
+
+    if ( pos == string::npos ) // No non-whitespace character in line
+        return false;
+
+    if ( pos == 0 )
+        return boost::starts_with( line, comment_marker );
+    else
+        return boost::starts_with( line.substr( pos ), comment_marker );
+}
+
+
+bool CommentedConfigFile::is_empty_line( const string & line )
+{
+    if ( line.empty() )
+        return true;
+
+    return line.find_first_not_of( WHITESPACE ) == string::npos;
+}
+
+
+void CommentedConfigFile::split_off_comment( const string & line,
+					     string & content_ret,
+					     string & comment_ret )
+{
+    size_t pos = line.find( comment_marker );
+
+    if ( pos == string::npos )
+    {
+        comment_ret = "";
+        content_ret = line;
+    }
+    else
+    {
+        content_ret = line.substr( 0, pos-1 );
+        comment_ret = line.substr( pos );
+    }
+
+    strip_trailing_whitespace( content_ret );
+}
+
+
+void CommentedConfigFile::strip_trailing_whitespace( string & line )
+{
```

Comment:
Yes, I looked at that, and the resulting code was a lot less clear than this, so I preferred this own implementation.